### ISSUE TITLE
deleted_after_time not working properly

### ISSUE THREAD

**Issue by ajitdsa** [ActsAsParanoid/acts_as_paranoid#42]
Hi guys, I just installed the latest acts_as_paranoid and was trying to test out deleted_after_time.

```
irb(main):002:0> Item.deleted_after_time(2.years.ago)
  Item Load (0.7ms)  SELECT "items".* FROM "items" WHERE ("items"."deleted_at" IS NULL) AND (deleted_at > '2014-01-05 22:05:30.919562')
=> #<ActiveRecord::Relation []>
```

There are definitely items that match, but it seems like the 1st WHERE condition is causing the problem. I don't think we need it. Am I missing something?

**Comment by ajitdsa** [comment]
Hmm, actually I see in the core.rb file that the only_deleted method tries to look at string_type_with_deleted_value? to determine if it should use IS NOT NULL. I just looked in my database and the deleted_at column is a timestamp field, not a string. Is there somewhere else I should be checking?

**Comment by zzak** [comment]
`deleted_at` should default to a `time` field anyways.

Based on this `only_deleted` should be using "IS NOT NULL"

@ajitdsa Are you still running into this problem? I just merged #34 which might be related.

**Comment by mvz** [comment]
I can still reproduce this.
